### DISCUSSION TITLE
Nerdctl can't use locally built images in Dockerfile's

### DISCUSSION THREAD

**Discussion by zagaria** [containerd/nerdctl#1443] [Q&A] [answered]
### Description

Nerdctl can't use locally built images, every time trying to pull image from remote source.

### Steps to reproduce the issue

1. Create Dockerfile 
```
FROM alpine:3.16

RUN apk update && apk add ca-certificates curl
```
2. Build ```nerdctl build -t my-image:1.0.0 .```
3. Check images using ```nerdctl images```
4. Create new file ```test.Dockerfile``` that will use previously created image
```
FROM my-image:1.0.0

RUN apk update && apk add openssl
```
5. Try to build ```nerdctl build -t my-image-2:1.0.0 -f test.Dockerfile .```

### Describe the results you received and expected

Received:

```
[+] Building 0.9s (3/3) FINISHED                                                                                                                                                                                   
 => [internal] load build definition from test.Dockerfile                                                                                                                                                     0.0s
 => => transferring dockerfile: 97B                                                                                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                                                                             0.0s
 => => transferring context: 2B                                                                                                                                                                               0.0s
 => ERROR [internal] load metadata for docker.io/library/my-image:1.0.0                                                                                                                                       0.9s
------
 > [internal] load metadata for docker.io/library/my-image:1.0.0:
------
test.Dockerfile:1
--------------------
   1 | >>> FROM my-image:1.0.0
   2 |     
   3 |     RUN apk update && apk add openssl
--------------------
error: failed to solve: my-image:1.0.0: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
FATA[0001] unrecognized image format
```

Expected: image successfully built

### What version of nerdctl are you using?

master, v0.23

**Comment by AkihiroSuda** [comment]
Please make sure to configure BuildKit to use containerd worker: https://github.com/containerd/nerdctl/blob/v0.23.0/docs/build.md#setting-up-buildkit-with-containerd-worker

e.g.,
```
CONTAINERD_NAMESPACE=default containerd-rootless-setuptool.sh install-buildkit-containerd
```

**Comment by zagaria** [comment, marked as answer]
Hi @AkihiroSuda,

Thank you for the quick response, I will review it.

But I'm using the official docker package https://github.com/containerd/nerdctl/pkgs/container/nerdctl - does your answer mean that described issue is expected behaviour for nerdctl docker image https://github.com/containerd/nerdctl/pkgs/container/nerdctl/41194079?tag=v0.23.0 and need to prepare own image?

**Reply by AkihiroSuda** [reply]
Fix is in progress: https://github.com/containerd/nerdctl/pull/1455

**Reply by AkihiroSuda** [reply]
Fixed in `ghcr.io/containerd/nerdctl:v1.0.0`

**Reply by zagaria** [reply]
Tested, works.

Thank you!

**Comment by cielavenir** [comment]
I have similar issue, so I filed https://github.com/containerd/nerdctl/issues/2550

**Reply by cielavenir** [reply]
So, setting namespace="default" caused "lease not found" error in my environment.
